java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -0,0 +1,1402 @@
+   *                                      max allowed
+   * @throws OutOfMemoryException if the internal memory allocation fails
+   */
+  public void reallocViewBuffer(long desiredAllocSize) {
+    if (desiredAllocSize == 0) {
+      return;
+    }
+    long newAllocationSize = CommonUtil.nextPowerOfTwo(desiredAllocSize);
+    assert newAllocationSize >= 1;
+
+    checkDataBufferSize(newAllocationSize);
+    // for each set operation, we have to allocate 16 bytes
+    // here we are adjusting the desired allocation-based allocation size
+    // to align with the 16bytes requirement.
+    newAllocationSize = roundUpToMultipleOf16(newAllocationSize);
+
+    final ArrowBuf newBuf = allocator.buffer(newAllocationSize);
+    newBuf.setBytes(0, viewBuffer, 0, viewBuffer.capacity());
+
+    viewBuffer.getReferenceManager().release();
+    viewBuffer = newBuf;
+    lastValueAllocationSizeInBytes = viewBuffer.capacity();
+  }
+
+  /**
+   * Reallocate the data buffer for views.
+   *
+   * @param desiredAllocSize allocation size in bytes
+   */
+  public void reallocViewDataBuffer(long desiredAllocSize) {
+    if (desiredAllocSize == 0) {
+      return;
+    }
+
+    if (dataBuffers.isEmpty()) {
+      return;
+    }
+
+    ArrowBuf currentBuf = dataBuffers.get(dataBuffers.size() - 1);
+    if (currentBuf.capacity() - currentBuf.writerIndex() >= desiredAllocSize) {
+      return;
+    }
+
+    final long newAllocationSize = CommonUtil.nextPowerOfTwo(desiredAllocSize);
+    assert newAllocationSize >= 1;
+
+    checkDataBufferSize(newAllocationSize);
+
+    final ArrowBuf newBuf = allocator.buffer(newAllocationSize);
+    dataBuffers.add(newBuf);
+  }
+
+  /**
+  *  Reallocate Validity buffer.
+  */
+  public void reallocValidityBuffer() {
+    int targetValidityCount = capAtMaxInt((validityBuffer.capacity() * 8) * 2);
+    if (targetValidityCount == 0) {
+      if (lastValueCapacity > 0) {
+        targetValidityCount = lastValueCapacity;
+      } else {
+        targetValidityCount = 2 * INITIAL_VALUE_ALLOCATION;
+      }
+    }
+
+    long validityBufferSize = computeValidityBufferSize(targetValidityCount);
+
+    final ArrowBuf newValidityBuffer = allocator.buffer(validityBufferSize);
+    newValidityBuffer.setBytes(0, validityBuffer, 0, validityBuffer.capacity());
+    newValidityBuffer.setZero(validityBuffer.capacity(), newValidityBuffer.capacity() - validityBuffer.capacity());
+    validityBuffer.getReferenceManager().release();
+    validityBuffer = newValidityBuffer;
+
+    lastValueCapacity = getValueCapacity();
+  }
+
+  private long computeValidityBufferSize(int valueCount) {
+    return (valueCount + 7) / 8;
+  }
+
+  /**
+   * Get the size (number of bytes) of underlying view buffer.
+   * @return number of bytes in the view buffer
+   */
+  @Override
+  public int getByteCapacity() {
+    return capAtMaxInt(viewBuffer.capacity());
+  }
+
+  @Override
+  public int sizeOfValueBuffer() {
```

Comment:
This operation is ill-defined, I think. It's not clear to me what it's supposed to actually mean for this type.

Reply:
Thinking about this again... 
The interface definition for this function is 

```java
/**
   * Provide the number of bytes contained in the valueBuffer.
   * @return the number of bytes in valueBuffer.
   */
   int sizeOfValueBuffer();
```
We don't have a `valueBuffer` for views. I think what we could do is the following. 

First step, Check usage and avoid. This is only used in test cases and that is used with the abstracted test cases using the common interface to minimize test code duplication for views and non-views. The test case is trivial it just needs to know the size of the data stored. `sizeOfValueBuffer` usage can be avoided and we can just use the `getValueLength` and `getValueCount` functions to get the same functionality. 

In terms of functionality support, we should probably override this and put `UnsupportedOperationException`. And we can introduce a separate function to get the `number of bytes contained in the viewBuffer`, which is `16* valueCount`.

WDYT?